scripts/gatkcondaenv.yml.template

```diff
@@ -1,50 +1,61 @@
 # $condaEnvDescription
 #
+# Only update this environment if there is a *VERY* good reason to do so!
+# If the build is broken but could be fixed by doing something else, then do that thing instead.
+# Ensuring the correct environment for canonical (or otherwise reasonable) usage of our standard Docker takes precedence over edge cases.
+# If you break the environment, you are responsible for fixing it and also owe the last developer who left this in a reasonable state a beverage of their choice.
+# (This may be yourself, and you'll appreciate that beverage while you tinker with dependencies!)
+#
+# When changing dependencies or versions in this file, check to see if the "supportedPythonPackages" DataProvider
+# used by the testGATKPythonEnvironmentPackagePresent test in PythonEnvironmentIntegrationTest needs to be updated
+# to reflect the changes.
+#
 name: $condaEnvName
 channels:
+# if channels other than conda-forge are added and the channel order is changed (note that conda channel_priority is currently set to flexible),
+# verify that key dependencies are installed from the correct channel and compiled against MKL
+- conda-forge
 - defaults
 dependencies:
-- certifi=2016.2.28=py36_0
-- intel-openmp=2018.0.0
-- mkl=2018.0.1
-- mkl-service=1.1.2
-- openssl=1.0.2l=0
-- pip=9.0.1=py36_1
-- python=3.6.2=0
-- readline=6.2=2
-- setuptools=36.4.0=py36_1
-- sqlite=3.13.0=0
-- anaconda::tensorflow=1.12.0=mkl_py36h69b6ba0_0
-- tk=8.5.18=0
-- wheel=0.29.0=py36_0
-- xz=5.2.3=0
-- zlib=1.2.11=0
+
+# core python dependencies
+- conda-forge::python=3.6.10        # do not update
+- pip=20.0.2                        # specifying channel may cause a warning to be emitted by conda
+- conda-forge::mkl=2019.5           # MKL typically provides dramatic performance increases for theano, tensorflow, and other key dependencies
+- conda-forge::mkl-service=2.3.0
+- conda-forge::numpy=1.17.5         # do not update, this will break scipy=0.19.1
+                                    #   verify that numpy is compiled against MKL (e.g., by checking *_mkl_info using numpy.show_config())
+                                    #   and that it is used in tensorflow, theano, and other key dependencies
+- conda-forge::theano=1.0.4         # it is unlikely that new versions of theano will be released
+                                    #   verify that this is using numpy compiled against MKL (e.g., by the presence of -lmkl_rt in theano.config.blas.ldflags)
+- defaults::tensorflow=1.15.0       # update only if absolutely necessary, as this may cause conflicts with other core dependencies
+                                    #   verify that this is using numpy compiled against MKL (e.g., by checking tensorflow.pywrap_tensorflow.IsMklEnabled())
```

Comment:
Should we actually run this command during the base image build and fail if it's not working?

Reply:
I thought about adding some stuff to the package version checks in the unit tests, but it's a good idea to fail early at the image build.  So I should add something to the bash script for building the base image?

I'll have to look at the message output by the numpy MKL check, it's not as simple to parse as the one generated by this tensorflow check.